crypto/bls12381/key_bls12381.go

```diff
@@ -37,7 +39,20 @@ var _ crypto.PrivKey = &PrivKey{}
 
 type PrivKey []byte
 
-// NewPrivateKeyFromBytes build a new key from the given bytes.
+// NewPrivateKeyFromBytes generates a new random key using `secret` for the seed
+// TODO simplify once `github.com/cosmos/crypto/` has `GenPrivKeyFromSeed`
+func GenPrivKeyFromSecret(secret []byte) (PrivKey, error) {
+	seed := sha256.Sum256(secret) // We need 32 bytes
+
+	// Defensive check: have we generated a valid secret key?
+	secretKey := blst.KeyGen(seed[:]).Serialize()
+	if bls12381.IsZero(secretKey) {
+		return nil, errors.New("generated secret key is zero")
+	}
+	return secretKey, nil
```

Comment:
```suggestion
	return secretKey.Marshal(), nil
```

Do we need to marshal the secret key similar to how we do it in `NewPrivateKeyFromBytes` and `GenPrivKey`?

Reply:
I think so. Please check the simplified version if you have the chance. It does the `.Marshall()` as you suggest here